api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -65,7 +65,7 @@
 	 */
 	@Authorized( { PrivilegeConstants.EDIT_USER_PASSWORDS })
 	@Logging(ignoredArgumentIndexes = { 1, 2 })
-	public void changePassword(User user, String oldPassword, String newPassword) throws APIException;
+	public void changePassword(User user, String secret, String newPassword) throws APIException;
```

Comment:
The new method will have the same signature as `public void changePassword(String pw, String pw2)` so I just thought and have that method take either `secret answer` or `old password` and work either way.